### PR TITLE
Routed usage consent dialog strings, image, about box title, start page image via resource provider

### PR DESCRIPTION
**Pull request by aparajit-pratap**
ping @Benglin

### CODE REVIEW

File: src/DynamoCoreWpf/Interfaces/IBrandingResourceProvider.cs [hunk 1]
```diff
@@ -6,12 +6,24 @@ public enum ResourceName
...
+    }
+
+    public enum UsageConsentFormStringResource
+    {
+        UsageConsentFormTitle,
+        UsageConsentFormMessage1,
+        UsageConsentFormFeatureUsage,
+        UsageConsentFormNodeUsage,
+        UsageConsentFormMessage2,
+        UsageConsentFormConsent
```
Comment (Benglin) [inline src/DynamoCoreWpf/Interfaces/IBrandingResourceProvider.cs]: As discussed:

```csharp
struct ResourceNames
{
    enum AboutBox
    {
        Title,   // String
        MainIcon // ImageSource
    }
    
    enum ConsentDialog
    {
        AgreementOne, // String
        AgreementTwo  // String
    }
}

public interface IBrandingResourceProvider
{
    ImageSource GetImageSource(ResourceNames.AboutBox name);

    string GetString(ResourceNames.AboutBox name);
    string GetString(ResourceNames.ConsentDialog name);
}
```

**Comment by aparajit-pratap** [comment]
@Benglin amended as discussed!

File: src/DynamoCoreWpf/Controls/StartPage.xaml.cs [hunk 2]
```diff
@@ -414,6 +414,8 @@ private void OnStartPageLoaded(object sender, RoutedEventArgs e)
             this.codeListBox.ItemsSource = startPageViewModel.ContributeLinks;
             this.recentListBox.ItemsSource = startPageViewModel.RecentFiles;
             this.sampleFileTreeView.ItemsSource = startPageViewModel.SampleFiles;
+
+            StartPageLogo.Source = dynamoViewModel.BrandingResourceProvider.GetImageSource(Wpf.Interfaces.ResourceNames.StartPage.Image);
```
Comment (Benglin) [inline src/DynamoCoreWpf/Controls/StartPage.xaml.cs]: Too long a line...

```csharp
var id = Wpf.Interfaces.ResourceNames.StartPage.Image;
StartPageLogo.Source = dynamoViewModel.BrandingResourceProvider.GetImageSource(id);
```

File: src/DynamoCoreWpf/Views/Core/DynamoView.xaml [hunk 3]
```diff
@@ -769,7 +769,6 @@
                                   Header="{x:Static p:Resources.DynamoViewDebugMenuDumpLibrary}"
                                   Command="{Binding DumpLibraryToXmlCommand}"
                                   IsEnabled="True" />
->>>>>>> 548ac9bc53b706ac3418e48fbae443b713346a50:src/DynamoCoreWpf/Views/DynamoView.xaml
```
Comment (Benglin) [inline src/DynamoCoreWpf/Views/Core/DynamoView.xaml]: Now I know where did that come from, forgot to tell you, but thanks for fixing!

File: src/DynamoCoreWpf/UI/DefaultBrandingResourceProvider.cs [hunk 4]
```diff
@@ -8,17 +8,32 @@ namespace Dynamo.Wpf.UI
...
                     break;
-                case ResourceName.UsageConsentFormImage:
+
+            }
+            if (image == null)
+            {
+                throw new InvalidEnumArgumentException(
+                    String.Format("Resource name not handled: {0}", resourceName));
+            }
+            return image;
```
Comment (Benglin) [inline src/DynamoCoreWpf/UI/DefaultBrandingResourceProvider.cs]: This is now repeating, can we do this?

```csharp
public ImageSource GetImageSource(ResourceNames.AboutBox resourceName)
{
    // Bla bla bla...

    return EnsureImageLoaded(image, resourceName.ToString());
}

private ImageSource EnsureImageLoaded(ImageSource image, string name)
{
    if (image != null)
        return image;

    throw new InvalidEnumArgumentException(
        String.Format("Resource name not handled: {0}", name));
}
```

**Comment by Benglin** [comment]
Thanks @aparajit-pratap, I'm going to check back on the other side of things to ensure they are in-sync, then after you addressed the above I'll merge it in.

**Comment by aparajit-pratap** [comment]
Done @Benglin

**Comment by Benglin** [comment]
Thanks @aparajit-pratap, way to go! LGTM

**Pull request merged**